### ISSUE TITLE
docker build error: Cannot find module 'semver'

### ISSUE THREAD

**Issue by fgiorlando** [HospitalRun/hospitalrun-frontend#1045]
When attempting to build with the latest hospitalrun-frontend from github, I encounter the following error.

(Docker version 17.03.1-ce, build c6d412e on Ubuntu 14.04)

```
$ docker build -t hospitalrun-frontend .
```

```
...
module.js:471
    throw err;
    ^

Error: Cannot find module 'semver'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/lib/node_modules/npm/lib/utils/unsupported.js:2:14)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
```

Looks like it may be related to the following npm issue: https://github.com/npm/npm/issues/15611

**Comment by fgiorlando** [comment]
Still trying to get this working, so far the following haven't worked:

0) deleted all docker containers and started fresh
1) tried node 7.x
```
RUN curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
```
2) tried the following modified command:
```
RUN curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -

# added
RUN apt-get update && apt-get install -y nodejs
RUN npm install npm@latest -g
```
3) tried a separate semver command
```
RUN npm install -g semver
```

the error remains the same as in the previous comment.
Also noticed this similar bug report: https://github.com/npm/npm/issues/15558

Is anyone else having trouble building the docker image?

**Comment by fgiorlando** [comment]
OK, I fixed this error by getting rid of the npm update line (as described in this post: https://github.com/npm/npm/issues/15558)

I think the npm update causes a mismatch between the npm version and the cached modules, but I'm not really sure if that's the cause.
So I'd consider this more of a workaround than a fix!

changes to `Dockerfile`:

```
- RUN npm install -g npm && npm install -g ember-cli@latest && npm install -g bower
+ #RUN npm install -g npm@latest
+ RUN npm install -g ember-cli@latest
+ RUN npm install -g bower
```

**Comment by nfung** [comment]
I got the same error and was able to fix it by moving it inline. After that was fixed, I encountered several other issues - latest issue i'm still working on is that the app is not able to connect to the db.

**Comment by fgiorlando** [comment]
Good to hear I'm not the only one.
Regarding the database connection issue, it would be good to start another issue...

**Comment by baoqchau** [comment]
Hello @fgiorlando. I tried to run the project with docker build and have successfully ran it. I used Ubuntu 16.10, Docker version 17.06.0-ce, build 02c1d87, and node version 7.7.1. From my experience, the solution when we encountered `Error: Cannot find module` is that we should delete `node_modules` folder and run `npm install` (and `bower install`, for Hospitalrun specifically) again. Hope that this solve your problem.
